subsys/bluetooth/controller/ll_sw/ull_llcp_enc.c

```diff
@@ -204,6 +232,37 @@ static void lp_enc_send_enc_req(struct ll_conn *conn, struct proc_ctx *ctx, uint
 	}
 }
 
+static void lp_enc_send_pause_enc_req(struct ll_conn *conn, struct proc_ctx *ctx, uint8_t evt, void *param)
+{
+	if (!tx_alloc_is_available()) {
+		ctx->state = LP_ENC_STATE_WAIT_TX_PAUSE_ENC_REQ;
+	} else {
+		lp_enc_tx(conn, ctx, PDU_DATA_LLCTRL_TYPE_PAUSE_ENC_REQ);
+		/* Wait for the LL_PAUSE_ENC_RSP */
+		ctx->rx_opcode = PDU_DATA_LLCTRL_TYPE_PAUSE_ENC_RSP;
+		ctx->state = LP_ENC_STATE_WAIT_RX_PAUSE_ENC_RSP;
+	}
+}
+
+static void lp_enc_send_pause_enc_rsp(struct ll_conn *conn, struct proc_ctx *ctx, uint8_t evt, void *param)
+{
+	if (!tx_alloc_is_available()) {
+		ctx->state = LP_ENC_STATE_WAIT_TX_PAUSE_ENC_RSP;
+	} else {
+		lp_enc_tx(conn, ctx, PDU_DATA_LLCTRL_TYPE_PAUSE_ENC_RSP);
```

Comment:
My understanding: this is a peripheral device responding to enc_pause. And this call enqueues the pdu for tx, which should be done encrypted?

Thus is appears off to me that in line#259 the enc_tx = 0U - should this not only happen after the above tx is acked?